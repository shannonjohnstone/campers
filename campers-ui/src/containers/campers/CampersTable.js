import React from 'react';
import { array } from 'prop-types';
import { Table } from 'react-bootstrap';
import Camper from './Camper';

const CampersTable = ({ campers = [] }) => {
  return (
    <Table striped bordered hover responsive size="sm">
      <thead>
        <tr>
          <th>Ranking</th>
          <th>Avatar</th>
          <th>Username</th>
          <th>Monthly Total</th>
          <th>All Time Total</th>
        </tr>
      </thead>
      <tbody>
        {campers.map((camper, index) => (
          <Camper
            key={camper.username}
            camper={{ ...camper, rank: index + 1 }}
          />
        ))}
      </tbody>
    </Table>
  );
};

CampersTable.propTypes = {
  campers: array,
};

export default CampersTable;
