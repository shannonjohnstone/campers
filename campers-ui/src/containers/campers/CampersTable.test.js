import React from 'react';
import CampersTable from './CampersTable';
import { render } from 'react-testing-library';

const mockCamper = [
  {
    id: '5c79ef86802e9c20cc0d52ce',
    img: 'https://source.unsplash.com/random/32x32',
    username: 'Luann Ballard',
    avatar:
      'https://www.tinygraphs.com/squares/859?theme=bythepool&numcolors=1&size=32&fmt=svg',
    monthly_total: 55,
    total: 3543,
    rank: 1,
  },
  {
    id: '5c79ef86dd148bb0a5dd664c',
    img: 'https://source.unsplash.com/random/32x32',
    username: 'Lawrence Chaney',
    avatar:
      'https://www.tinygraphs.com/squares/595?theme=duskfalling&numcolors=7&size=32&fmt=svg',
    monthly_total: 96,
    total: 457,
    rank: 2,
  },
];

describe('CampersTable component', () => {
  test('Component snapshot', () => {
    const { container } = render(<CampersTable camper={mockCamper} />);
    expect(container).toMatchSnapshot();
  });
});
