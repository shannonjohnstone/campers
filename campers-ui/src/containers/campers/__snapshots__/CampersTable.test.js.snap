// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampersTable component Component snapshot 1`] = `
<div>
  <div
    class="table-responsive"
  >
    <table
      class="table table-sm table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Ranking
          </th>
          <th>
            Avatar
          </th>
          <th>
            Username
          </th>
          <th>
            Monthly Total
          </th>
          <th>
            All Time Total
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
